File: src/dere_productivity/scripts/taskwarrior_mcp_wrapper.py
#!/usr/bin/env python3
"""Wrapper to run taskwarrior MCP server with resolved path."""
import subprocess
import sys
from pathlib import Path


def main():
    package_root = Path(__file__).resolve().parent.parent
    mcp_server = package_root / "mcp-server" / "dist" / "index.js"

    if not mcp_server.exists():
        print(f"Error: MCP server not found at {mcp_server}", file=sys.stderr)
        print("Run 'just build-mcp' to build it", file=sys.stderr)
        sys.exit(1)

    # Run the MCP server
    subprocess.run(["node", str(mcp_server)] + sys.argv[1:])

if __name__ == "__main__":
    main()

File: src/dere_plugins/dere_productivity/scripts/taskwarrior_mcp_wrapper.ts
import { existsSync } from "node:fs";
import { spawnSync } from "node:child_process";
import { join, resolve } from "node:path";

function main(): void {
  const packageRoot = resolve(import.meta.dir ?? ".", "..");
  const mcpServer = join(packageRoot, "mcp-server", "dist", "index.js");

  if (!existsSync(mcpServer)) {
    console.error(`Error: MCP server not found at ${mcpServer}`);
    console.error("Run 'just build-mcp' to build it");
    process.exit(1);
  }

  const result = spawnSync("node", [mcpServer, ...process.argv.slice(2)], {
    stdio: "inherit",
  });
  process.exit(result.status ?? 1);
}

if (import.meta.main) {
  main();
}
